.github/workflows/push.yml

```diff
@@ -41,6 +44,13 @@ jobs:
         uses: ./.github/actions/setup-oracle
       - name: apt
         uses: ./.github/actions/apt-x64
+      - name: ccache
+        uses: hendrikmuhs/ccache-action@v1.2
+        with:
+          # This duplicates the "job.name" expression above because
+          # GitHub has no way to query the job name (github.job is the
+          # job id, not the job name)
+          key: "LINUX_X64_${{ matrix.debug && 'DEBUG' || 'RELEASE' }}_${{ matrix.zts && 'ZTS' || 'NTS' }}-${{hashFiles('main/php_version.h')}}"
```

Comment:
```suggestion
          key: "${{ github.workflow }}-${{ github.job }}-${{ toJSON(matrix) }}-${{ hashFiles('main/php_version.h') }}"
```

Reply:
That's better because it doesn't duplicate the expression, but it's worse because a hash is more obscure. I don't have a real opinion on what to choose.

Reply:
no hash added and json is quite readable :)

Reply:
Ah, it's JSON, not hash of a JSON ... yes, that makes sense.

Reply:
Doesn't work:
```
  Error: Restoring cache failed: ValidationError: Key Validation Error: ccache-Push-LINUX_X64-{
    "debug": true,
    "zts": false
  }-10c280ee69e81a8f532bab2756a061cccc70829ec83708c23eb95d1f14f49c53- cannot contain commas.
```

Reply:
Hash of JSON would solve that, but no hash function exists :( :) What about moving the ccache step into separate file and calculating the cache key there using bash?

see https://github.com/atk4/core/blob/develop/.github/workflows/test-unit.yml l37 and l42 for how data can be passed between steps, ie. from bash step to GH action step

Reply:
I don't think that's worth the complexity.